tests/regression/src/lib.rs

```diff
@@ -0,0 +1,143 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use s2n_tls::{config::Builder, security, testing::{CertKeyPair, InsecureAcceptAllCertificatesHandler}};
+type Error = s2n_tls::error::Error;
+
+
+
+// Function to create default config with specified parameters
+pub fn set_config(
+    cipher_prefs: &security::Policy,
+    keypair: CertKeyPair
+) -> Result<s2n_tls::config::Config, Error> {
+    let mut builder = Builder::new();
+    builder
+        .set_security_policy(cipher_prefs)
+        .expect("Unable to set config cipher preferences");
+    builder
+        .set_verify_host_callback(InsecureAcceptAllCertificatesHandler {})
+        .expect("Unable to set a host verify callback.");
+    builder
+        .load_pem(keypair.cert(), keypair.key())
+        .expect("Unable to load cert/pem");
+    builder.trust_pem(keypair.cert()).expect("load cert pem");
+    builder.build()
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use crabgrind as cg;
+    use s2n_tls::testing::TestPair;
+    use std::env;
+    use std::process::Command;
+
+    const COST: u64 = 1_000_000; //configurable threshold for regression
+    // environment variable to determine whether to run under valgrind or solely test functionality
+    fn is_running_under_valgrind() -> bool {
+        env::var("VALGRIND").is_ok()
+    }
```

Comment:
this environment variable seems poorly named for what it's doing

Reply:
I'll go with ENABLE_VLAGRIND, which makes more sense at least from the user invoking it perspective. The actual way this variable is used via recursive calls to the test is hard to name because only when it is not set in the recursive call is actually when it runs under valgrind.